### PR TITLE
Filter modal hooks tests

### PR DESCRIPTION
**Pull request by ranquild**
Parametrized unit tests for shared hooks between `FilterPicker` and `FilterModal`.

### CODE REVIEW

File: frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+      const newOperator = ">";
+      setOperator(newOperator);
+      setValues(getDefaultValues(newOperator, values));
+    });
+
+    act(() => {
+      const { operator, values, getFilterClause } = result.current;
+      const newFilter = checkNotNull(getFilterClause(operator, values));
+
+      expect(
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts:282]: afaik `expect` should go outside `act` as there is no any side effect in it

File: frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+          stageIndex,
+          column,
+        }),
+      );
+
+      act(() => {
+        const { getFilterClause } = result.current;
+        const newFilter = checkNotNull(getFilterClause(operator, values));
+
+        expect(
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts:209]: move expect outside `act`

File: frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+      act(() => {
+        const { setValues } = result.current;
+        setValues(values);
+      });
+
+      act(() => {
+        const { operator, values, getFilterClause } = result.current;
+        const newFilter = checkNotNull(getFilterClause(operator, values));
+
+        expect(Lib.displayInfo(query, stageIndex, newFilter)).toMatchObject({
```
Comment (uladzimirdev) [inline frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts:181]: move `expect` outside act

**Review by uladzimirdev — APPROVED** [review]
LGTM, I'd only move `expect` from `act` everywhere

File: frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/querying/hooks/use-time-filter/use-time-filter.unit.spec.ts]: Fixed

**Pull request merged**